reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -912,6 +913,28 @@ public static <T, D> Mono<T> using(Callable<? extends D> resourceSupplier,
 		return using(resourceSupplier, sourceSupplier, resourceCleanup, true);
 	}
 
+	/**
+	 * Uses a {@link AutoCloseable} resource, generated by a supplier for each individual Subscriber,
+	 * while streaming the value from a Mono derived from the same resource and makes sure
+	 * the resource is released if the sequence terminates or the Subscriber cancels.
+	 * <p>
+	 * Unlike in {@link Flux#using(Callable, Function, Consumer) Flux}, in the case of a valued {@link Mono} the cleanup
+	 * happens just before passing the value to downstream. In all cases, exceptions raised by the cleanup
+	 * {@link Consumer} may override the terminal event, discarding the element if the derived {@link Mono} was valued.
+	 * <p>
+	 * <img class="marble" src="doc-files/marbles/usingForMono.svg" alt="">
+	 *
+	 * @param resourceSupplier a {@link Callable} that is called on subscribe to create the resource
+	 * @param sourceSupplier a {@link Mono} factory to create the Mono depending on the created resource
+	 * @param <T> emitted type
+	 * @param <D> resource type
+	 *
+	 * @return new {@link Mono}
+	 */
+	public static <T, D extends AutoCloseable> Mono<T> using(Callable<D> resourceSupplier,
+			Function<? super D, ? extends Mono<? extends T>> sourceSupplier) {
+		return using(resourceSupplier, sourceSupplier, Flux.AUTO_CLOSE, true);
```

Comment:
Instead of using `Flux.AUTO_CLOSE` here, perhaps it makes sense to move `AUTO_CLOSE` to the `reactor.core.Exceptions` class and use that from `Mono` and `Flux`.

Reply:
I wonder whether `Exceptions.propagate(t)` would be a good candidate here. It feels like a natural choice.

Reply:
✅ updated, nice suggestion!